server/middleware.ts

```diff
@@ -73,8 +73,18 @@ export function isUserOrAdmin(request: express.Request, response: express.Respon
 export function isAdmin(request: express.Request, response: express.Response, next: express.NextFunction) {
 	response.setHeader("Cache-Control", "private");
 	let user = request.user as IUser;
-	if (request.query.adminKey === config.secrets.adminKey) {
-		next();
+	const auth = request.headers.authorization;
+
+	if (auth && typeof auth === "string") {
+		const key = new Buffer(auth.split(" ")[1], "base64").toString();
```

Comment:
This will crash the server if the `Authorization` header doesn't have a space in it due to it indexing an array of length 0 or 1